crates/cow-amm/src/cache.rs

```diff
@@ -110,17 +226,21 @@ impl EventStoring<ethcontract::Event<CowAmmEvent>> for Storage {
     }
 
     async fn last_event_block(&self) -> anyhow::Result<u64> {
-        let cache = self.0.cache.read().await;
-
-        let last_block = cache
-            .last_key_value()
-            .map(|(block, _amms)| *block)
-            .unwrap_or(self.0.start_of_index);
-        Ok(last_block)
+        let mut ex = self.0.db.acquire().await?;
+        database::last_indexed_blocks::fetch(&mut ex, &self.0.factory_address.to_string())
+            .await?
+            .map(|block| block.try_into().context("last block is not u64"))
+            .unwrap_or(Ok(self.0.start_of_index))
     }
 
-    async fn persist_last_indexed_block(&mut self, _new_value: u64) -> anyhow::Result<()> {
-        // storage is only in-memory so we don't need to persist anything here
+    async fn persist_last_indexed_block(&mut self, latest_block: u64) -> anyhow::Result<()> {
+        let mut ex = self.0.db.acquire().await?;
+        database::last_indexed_blocks::update(
+            &mut ex,
+            &self.0.factory_address.to_string(),
+            i64::try_from(latest_block).context("last block is not u64")?,
```

Comment:
This error is confusing, the integer *may* be a u64 but it's bigger than `1 << 32 - 1` (the positive side of the integer)